File: k-random-core/src/main/java/io/github/krandom/randomizers/range/MonthDayRangeRandomizer.java
package io.github.krandom.randomizers.range;

import java.time.MonthDay;
import java.time.Year;

public class MonthDayRangeRandomizer extends AbstractRangeRandomizer<MonthDay> {
  private final LocalDateRangeRandomizer localDateRangeRandomizer;

  public MonthDayRangeRandomizer(final MonthDay min, final MonthDay max) {
    super(min, max);
    localDateRangeRandomizer =
        new LocalDateRangeRandomizer(
            this.min.atYear(Year.now().getValue()), this.max.atYear(Year.now().getValue()));
  }

  public MonthDayRangeRandomizer(final MonthDay min, final MonthDay max, final long seed) {
    super(min, max, seed);
    localDateRangeRandomizer =
        new LocalDateRangeRandomizer(
            this.min.atYear(Year.now().getValue()), this.max.atYear(Year.now().getValue()), seed);
  }

  @Override
  protected void checkValues() {
    if (min.isAfter(max)) {
      throw new IllegalArgumentException("max must be after min");
    }
  }

  @Override
  protected MonthDay getDefaultMinValue() {
    return MonthDay.of(1, 1);
  }

  @Override
  protected MonthDay getDefaultMaxValue() {
    return MonthDay.of(12, 31);
  }

  @Override
  public MonthDay getRandomValue() {
    return MonthDay.from(localDateRangeRandomizer.getRandomValue());
  }
}

File: core/src/main/java/io/github/krandom/randomizers/range/MonthDayRangeRandomizer.kt
package io.github.krandom.randomizers.range

import java.time.MonthDay
import java.time.Year
import kotlin.random.Random

private const val JANUARY = 1
private const val DECEMBER = 12
private const val FIRST = 1
private const val THIRTY_FIRST = 31

/** Generate a random [MonthDay] within the given range. */
class MonthDayRangeRandomizer
/**
 * Create a new [MonthDayRangeRandomizer].
 *
 * @param min min value (inclusive)
 * @param max max value (exclusive)
 * @param seed initial seed
 */
@JvmOverloads
constructor(min: MonthDay?, max: MonthDay?, seed: Long = Random.nextLong()) :
  AbstractRangeRandomizer<MonthDay>(min, max, seed) {
  override val defaultMinValue: MonthDay
    get() = MonthDay.of(JANUARY, FIRST)

  override val defaultMaxValue: MonthDay
    get() = MonthDay.of(DECEMBER, THIRTY_FIRST)

  private val localDateRangeRandomizer: LocalDateRangeRandomizer =
    LocalDateRangeRandomizer(
      this.min.atYear(Year.now().value),
      this.max.atYear(Year.now().value),
      seed,
    )

  override fun getRandomValue(): MonthDay = MonthDay.from(localDateRangeRandomizer.getRandomValue())
}
